tool/generate_font.dart

```diff
@@ -60,22 +68,49 @@ void main(List<String> arguments) {
 
   generatedOutput.addAll(iconDefinitions.values);
 
-  generatedOutput.add('}');
+  generatedOutput.add('}\n');
+  generatedOutput.add('class FontAwesomeIconsPro {');
+  generatedOutput.add('  final String packageName;');
+  generatedOutput.add('  FontAwesomeIconsPro(this.packageName);\n');
+  generatedOutput.addAll(lightIconDefinitions.values);
+  generatedOutput.add('}\n');
 
   File output = new File('lib/font_awesome_flutter.dart');
   output.writeAsStringSync(generatedOutput.join('\n'));
 }
 
+const aliases = <String, String>{
+  '500px': 'fiveHundredPx',
+};
+
 String generateIconDefinition(String iconName, String style, String unicode) {
   style = '${style[0].toUpperCase()}${style.substring(1)}';
 
   String iconDataSource = 'IconData$style';
 
-  if (iconName == '500px') {
-    iconName = 'fiveHundredPx';
+  if (aliases.containsKey(iconName)) {
+    iconName = aliases[iconName];
   }
 
   iconName = new ReCase(iconName).camelCase;
 
   return 'static const IconData $iconName = const $iconDataSource(0x$unicode);';
 }
+
+String generateLightIconDefinition(String iconName, String unicode) {
+  if (aliases.containsKey(iconName)) {
+    iconName = aliases[iconName];
+  }
+
+  iconName = new ReCase(iconName).camelCase;
+
+  return '''
+      IconData _$iconName;
+      IconData get $iconName {
+        if(_$iconName == null) {
+          _$iconName = IconDataLight(packageName, 0x$unicode);
+        }
+        return _$iconName;
+      }
+    ''';
```

Comment:
I understand what you are trying to achieve, but this won't work out. The `FontAwesomeIconsPro` class will allocate memory for all defined icons whether they are used or not, only the getter methods will get tree-shaken (automatically removed). This results in a constant memory consumption of at least 1.2KB at all times (did a quick test to confirm the private attributes don't get removed also), up to 1.9KB (if every icon is used at least once). Additionally, I don't think icon data should be set at runtime, this is something that we know about before the program starts, so I recommend sticking with `static const`.

Reply:
Is <2KB that big of a deal? Modern devices usually have 2-6GB of RAM

Reply:
Why would I want to keep an object in memory that is literally "just there", if I only need _one_ object, which is literally 48 bytes? You cannot just assume that we have an unlimited amount of memory, flutter runs on devices with much less memory than what modern phones have.

Also, I regenerated `font_awesome_flutter.dart` with the pro `icons.json` and the memory consumption for **one** icon is **12.9KB**. Depending on how many icons you use, it can go up to **92KB(!)**, not including all getter methods, which will likely boost the memory footprint between 150 and 200KB...

Reply:
There's always splitting in in multiple files. If the Pro file isn't imported, it won't use any memory

Reply:
Of course, if you don't use the pro icons, you don't have anything to worry about. But this is about adding pro support, so we cannot assume that. Adding just one icon produces a 12.9KB object, instead of actually just delivering the 48B of static data that is needed.
The object can actually grow to more than 300KB if many items are used (rough calculation, including size of getter methods), which will only grow over time as more icons are being added. That is 1/3MB that just sits in memory and needs to be loaded during each start of the app.

Reply:
But if it's in it's dedicated pro file, which is never imported, it will get tree-shaken if you do not use the pro icons.

Reply:
You cannot argue "it's not bad if it's not being used" when this is about adding support for exactly that. The numbers I presented above are *with tree-shaking* applied.

Let's assume all icons are used and we used the `static const` method:
- 48B per icon
- 1648 light icons (we are not talking about *all* pro icons yet)
=> 7.9KB of memory used.

That is 3KB less than what the getter/private attribute method needs to display *one* icon!
And it only gets worse, for each icon used, it adds 48B for the icon data and 144B for the getter to the overall size. This doesn't sound like a big deal, but do the math: `(144B+48B)*1648 light icons = 316KB`. But let's assume we added all the pro icons to `FontAwesomeIconsPro`: `(144B+48B)*3831 pro icons = 735.5KB` (!) which is 4 times as much as what we would have needed with `static const`.

Obviously, noone would ever load *all* pro icons, but I hope you see my point.